encodings/dict/src/compute.rs

```diff
@@ -18,9 +29,32 @@ impl ArrayCompute for DictArray {
     fn take(&self) -> Option<&dyn TakeFn> {
         Some(self)
     }
+}
 
-    fn filter(&self) -> Option<&dyn FilterFn> {
-        Some(self)
+impl MaybeCompareFn for DictArray {
+    fn maybe_compare(
+        &self,
+        other: &ArrayData,
+        operator: Operator,
+    ) -> Option<VortexResult<ArrayData>> {
+        // If the RHS is constant, then we just need to compare against our encoded values.
+        if other
+            .statistics()
+            .get_as::<bool>(Stat::IsConstant)
```

Comment:
do you want a get or do you want a compute here?

Reply:
Almost always a get I think.

Reply:
Yeah, on ConstantArrays we set IsConstant in the constructor